File: www/src/lib/metastone/CardType.js
/**
 * Created by chris on 1/17/17.
 */
import EnumExtended from '../EnumExtended';

export default class CardType extends EnumExtended {
}

CardType.initEnum([
    'HERO',
    'MINION',
    'SPELL',
    'WEAPON',
    'HERO_POWER',
    'CHOOSE_ONE'
]);

File: spellsource-server/src/main/java/com/hiddenswitch/framework/graphql/CardType.java
package com.hiddenswitch.framework.graphql;

/**
 * ─── Enums ────────────────────────────────────────────────────
 */
public enum CardType {

    HERO("HERO"),
    MINION("MINION"),
    SPELL("SPELL"),
    WEAPON("WEAPON"),
    HERO_POWER("HERO_POWER"),
    GROUP("GROUP"),
    CHOOSE_ONE("CHOOSE_ONE"),
    ENCHANTMENT("ENCHANTMENT"),
    CLASS("CLASS"),
    FORMAT("FORMAT"),
    ROGUE_CHOICE("ROGUE_CHOICE");

    private final String graphqlName;

    private CardType(String graphqlName) {
        this.graphqlName = graphqlName;
    }

    @Override
    public String toString() {
        return this.graphqlName;
    }

}
